From 0a8d6bd2f945e87e83ffcb8a3e30e96797bdbbcc Mon Sep 17 00:00:00 2001
From: haiqiang yao <haiqiangx.yao@intel.com>
Date: Mon, 7 Mar 2016 13:22:27 +0800
Subject: [PATCH 3/5] atomisp2: Add bnlm parameter for isp2.7 pipe

this patch will add bnlm parameter for isp2.7 pipe

Depends on:https://icggerrit.ir.intel.com/#/c/107991/1

Change-Id: Ia1399810cb70af47574812d33fce64e308714b2c
Original-Tracked_On: #H1504179048
Signed-off-by: haiqiang yao <haiqiangx.yao@intel.com>
Reviewed-on: https://icggerrit.ir.intel.com/105955
Reviewed-by: Bin B Han <bin.b.han@intel.com>
Validation-Android: Wei Tao <wei.tao@intel.com>
Approver: Bin B Han <bin.b.han@intel.com>
Build: Dessert M BXT RVP (SC)
Integrator: WeiX Meng <weix.meng@intel.com>
Reviewed-by: WeiX Meng <weix.meng@intel.com>
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-25608
Signed-off-by: Tao Wei <wei.tao@intel.com>
Reviewed-on: https://android.intel.com:443/483550
---
 .../drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c | 13 +++++++++++++
 .../pci/atomisp2/atomisp_driver/atomisp_compat_css20.c      |  6 ++++++
 .../pci/atomisp2/atomisp_driver/atomisp_compat_css20.h      |  5 +++++
 .../pci/atomisp2/atomisp_driver/atomisp_compat_ioctl32.h    |  2 +-
 .../media/pci/atomisp2/atomisp_driver/atomisp_subdev.h      |  1 +
 5 files changed, 26 insertions(+), 1 deletion(-)

diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c
index 9334854..97279f0 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c
@@ -3330,6 +3330,9 @@ void atomisp_apply_css_parameters(
 	if (css_param->update_flag.xnr3_0_11_config)
 		atomisp_css_set_xnr3_0_11_config(asd, &css_param->xnr3_0_11_config);
 
+	if (css_param->update_flag.bnlm_config)
+		atomisp_css_set_bnlm_config(asd, &css_param->bnlm_config);
+
 	atomisp_css_set_isp_config_id(asd, css_param->isp_config_id);
 	/*
 	 * These configurations are on used by ISP1.x, not for ISP2.x,
@@ -3709,6 +3712,16 @@ int atomisp_cp_general_isp_parameters(struct atomisp_sub_device *asd,
 			(struct atomisp_xnr3_0_11_config *) &css_param->xnr3_0_11_config;
 	}
 
+	if (arg->bnlm_config && (from_user || !cur_config->bnlm_config)) {
+		if (copy_from_compatible(&css_param->bnlm_config,
+				arg->bnlm_config,
+				sizeof(struct atomisp_css_bnlm_config),
+				from_user))
+			return -EFAULT;
+		css_param->update_flag.bnlm_config =
+			(struct atomisp_bnlm_config *) &css_param->bnlm_config;
+	}
+
 	if (from_user)
 		css_param->isp_config_id = arg->isp_config_id;
 	/*
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_css20.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_css20.c
index b3825e5..033b120 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_css20.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_css20.c
@@ -3957,6 +3957,12 @@ void atomisp_css_set_xnr3_0_11_config(struct atomisp_sub_device *asd,
 	asd->params.config.xnr3_0_11_config = xnr3_0_11_config;
 }
 
+void atomisp_css_set_bnlm_config(struct atomisp_sub_device *asd,
+			struct atomisp_css_bnlm_config *bnlm_config)
+{
+	asd->params.config.bnlm_config = bnlm_config;
+}
+
 int atomisp_css_get_dp_config(struct atomisp_sub_device *asd,
 			struct atomisp_dp_config *config)
 {
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_css20.h b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_css20.h
index 826da05..84005ac 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_css20.h
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_css20.h
@@ -93,6 +93,8 @@
 #define atomisp_css_macc1_5_config	ia_css_macc1_5_config
 #define atomisp_css_macc1_5_table	ia_css_macc1_5_table
 #define atomisp_css_xnr3_0_11_config	ia_css_xnr3_0_11_config
+#define atomisp_css_bnlm_config	ia_css_bnlm_config
+
 
 #define CSS_PIPE_ID_PREVIEW	IA_CSS_PIPE_ID_PREVIEW
 #define CSS_PIPE_ID_COPY	IA_CSS_PIPE_ID_COPY
@@ -293,6 +295,9 @@ void atomisp_css_set_macc1_5_table(struct atomisp_sub_device *asd,
 void atomisp_css_set_xnr3_0_11_config(struct atomisp_sub_device *asd,
 			struct atomisp_css_xnr3_0_11_config *xnr3_0_11_config);
 
+void atomisp_css_set_bnlm_config(struct atomisp_sub_device *asd,
+			struct atomisp_css_bnlm_config *bnlm_config);
+
 unsigned int atomisp_css_debug_get_dtrace_level(void);
 
 int atomisp_css_debug_dump_isp_binary(void);
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_ioctl32.h b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_ioctl32.h
index 5a556a3..2fd74a4 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_ioctl32.h
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_ioctl32.h
@@ -239,7 +239,7 @@ struct atomisp_parameters32 {
 	compat_uptr_t macc1_5_config;
 	compat_uptr_t macc1_5_table;
 	compat_uptr_t xnr3_0_11_config;
-
+	compat_uptr_t bnlm_config;
 
 	compat_uptr_t	lin_2500_config;       /* Skylake: Linearization config */
 	compat_uptr_t	obgrid_2500_config;    /* Skylake: OBGRID config */
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_subdev.h b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_subdev.h
index 88d2e28..8574e03 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_subdev.h
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_subdev.h
@@ -198,6 +198,7 @@ struct atomisp_css_params {
 	struct ia_css_macc1_5_config macc1_5_config;
 	struct ia_css_macc1_5_table macc1_5_table;
 	struct ia_css_xnr3_0_11_config xnr3_0_11_config;
+	struct ia_css_bnlm_config bnlm_config;
 
 
 	/*
-- 
1.9.1

